DetectorDescription/DDCMS/plugins/DDDetectorESProducer.cc

```diff
@@ -66,7 +66,9 @@ class DDDetectorESProducer : public ESProducer, public EventSetupRecordIntervalF
 DDDetectorESProducer::DDDetectorESProducer(const ParameterSet& iConfig)
     : fromDB_(iConfig.getParameter<bool>("fromDB")),
       appendToDataLabel_(iConfig.getParameter<string>("appendToDataLabel")),
-      confGeomXMLFiles_(iConfig.getParameter<FileInPath>("confGeomXMLFiles").fullPath()),
+      confGeomXMLFiles_(iConfig.existsAs<FileInPath>("confGeomXMLFiles")
+                            ? iConfig.getParameter<FileInPath>("confGeomXMLFiles").fullPath()
+                            : "none"),
```

Comment:
Using the existence of parameters to control module behavior is generally considered as undesirable behavior. How about adding an explicit parameter to control whether the XML file is used or not?

Reply:
Nevermind, the parameter is already declared as optional in the `fillDescriptions()`.

Reply:
Actually, coming back. Guessing a bit the desired behavior based on the `fillDescriptions()`, if the `confGeomXMLFiles` is required when `fromDB == False`, how about
```cpp
confGeomXMLFiles_(fromDB_ ? "none" : iConfig.getParameter<FileInPath>("confGeomXMLFiles").fullPath()),
```
?

In this case the `fillDescriptions()` could also be enhanced along (untested)
```cpp
desc.ifValue(edm::ParameterDescription<bool>("fromDB", false, true),
            false >> edm::ParameterDescription<FileInPath>("confGeomXMLFiles", true));
```
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#A_Switch

Reply:
I'd think, the latter is not needed, because there are two configurations produced: one for DB and another for XML files. It seems clear which one needs the `confGeomXMLFiles`.